examples/merkle-tree-cli/src/main.rs

```diff
@@ -0,0 +1,97 @@
+mod commands;
+use clap::Parser;
+use commands::{MerkleArgs, MerkleEntity};
+use lambdaworks_crypto::{
+    hash::poseidon::Poseidon,
+    merkle_tree::{merkle::MerkleTree, proof::Proof},
+};
+use lambdaworks_math::{
+    elliptic_curve::short_weierstrass::curves::bls12_381::field_extension::BLS12381PrimeField,
+    field::element::FieldElement,
+};
+use std::{
+    fs::{self, File},
+    io::{self, BufWriter, Write},
+};
+
+type FE = FieldElement<BLS12381PrimeField>;
+
+fn load_fe_from_file(file_path: &String) -> Result<FE, io::Error> {
+    FE::from_hex(&fs::read_to_string(file_path)?)
+        .map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{:?}", e)))
+}
+
+fn load_tree_values(tree_path: &String) -> Result<Vec<FE>, io::Error> {
+    Ok(fs::read_to_string(tree_path)?
+        .lines()
+        .map(FE::from_hex_unchecked) // remove hex_unchecked for hex
+        .collect())
+}
+
+fn generate_merkle_tree(tree_path: String) -> Result<(), io::Error> {
+    let values: Vec<FE> = load_tree_values(&tree_path)?;
+
+    let merkle_tree = MerkleTree::<Poseidon<BLS12381PrimeField>>::build(&values);
+    let root = merkle_tree.root.representative().to_string();
+    println!("Generated merkle tree with root: {:?}", root);
+
+    let generated_tree_path = tree_path.replace(".csv", ".json");
+    let file = File::create(generated_tree_path)?;
+    let mut writer = BufWriter::new(file);
+    serde_json::to_writer_pretty(&mut writer, &merkle_tree)?;
+    println!("Saved tree to file");
+    Ok(())
+}
+
+fn generate_merkle_proof(tree_path: String, pos: usize) -> Result<(), io::Error> {
+    let values: Vec<FE> = load_tree_values(&tree_path)?;
+
+    let merkle_tree = MerkleTree::<Poseidon<BLS12381PrimeField>>::build(&values);
+
+    let Some(proof) = merkle_tree.get_proof_by_pos(pos) else {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Could not generate proof",
```

Comment:
This error could be a little better, maybe informing the user that the position was not found in the tree

Reply:
This one is a bit complicated, as the function `get_proof_by_pos` calls inside `build_merkle_path`, which panics the given index is out of bounds. 

https://github.com/lambdaclass/lambdaworks/blob/merkle-cli/crypto/src/merkle_tree/merkle.rs#L37-L58

Reply:
Modified `get_proof_by_pos` inside `merkle.rs` so it can cath an index out of bounds and return an `Err` if that happens.